### PR TITLE
Make retryOnConnectionFailure configurable

### PR DESCRIPTION
**Pull request by tgregory-block**
Chasing down intermittent failures in a service when calling an external API, and would like to be able to configure OkHttp to retry on connection failures.

### CODE REVIEW

**Review by tso — APPROVED** [review]

File: wisp-client/src/main/kotlin/wisp/client/HttpClientsConfig.kt [hunk 1]
```diff
@@ -101,7 +101,8 @@ data class HttpClientConfig(
     val keepAliveDuration: Duration? = null,
     val ssl: HttpClientSSLConfig? = null,
     val unixSocketFile: String? = null,
-    val protocols: List<String>? = null
+    val protocols: List<String>? = null,
+    val retryOnConnectFailure: Boolean? = null
```
Comment (tso) [inline wisp-client/src/main/kotlin/wisp/client/HttpClientsConfig.kt:105]: nit: retryOnConnectionFailure

Comment (tgregory-block) [reply wisp-client/src/main/kotlin/wisp/client/HttpClientsConfig.kt]: ty, fixed

File: wisp-client/src/main/kotlin/wisp/client/HttpClientFactory.kt [hunk 2]
```diff
@@ -21,7 +21,7 @@ class HttpClientFactory constructor(
     fun create(config: HttpClientEndpointConfig): OkHttpClient {
         // TODO(mmihic): Cache, proxy, etc
         val builder = unconfiguredClient.newBuilder()
-        builder.retryOnConnectionFailure(false)
+        builder.retryOnConnectionFailure(config.clientConfig.retryOnConnectionFailure ?: false)
         okHttpClientCommonConfigurator.configure(builder = builder, config = config)
```
Comment (mpeyper) [inline wisp-client/src/main/kotlin/wisp/client/HttpClientFactory.kt:25]: **suggestion (non-blocking):** I wonder if it makes sense to move this into `OkHttpClientCommonConfigurator` now?

**Review by mpeyper — APPROVED** [review]

File: wisp-client/src/main/kotlin/wisp/client/HttpClientFactory.kt (hunk 2 above)
Comment (tgregory-block) [reply wisp-client/src/main/kotlin/wisp/client/HttpClientFactory.kt]: Thanks, I completely missed that class. Fixed

**Review by mpeyper — APPROVED** [review]

**Review by chris-ryan-square — APPROVED** [review]

**Pull request merged**